components/uniset/src/enum_props.rs

```diff
@@ -4,69 +4,88 @@
 
 #![allow(clippy::upper_case_acronyms)]
 
+/// Selection constants for Unicode properties.
+/// These constants are used to select one of the Unicode properties.
+/// See UProperty in ICU4C.
 #[derive(Clone, PartialEq, Debug)]
 pub enum EnumeratedProperty {
-    BidiClass,
-    BidiPairedBracketType,
-    CanonicalCombiningClass,
-    DecompositionType,
-    EastAsianWidth,
-    GeneralCategory,
-    GraphemeClusterBreak,
-    HangulSyllableType,
-    IndicPositionalCategory,
-    IndicSyllabicCategory,
-    JoiningGroup,
-    JoiningType,
-    LineBreak,
-    LeadCanonicalCombiningClass,
-    NFCQuickCheck,
-    NFDQuickCheck,
-    NFKCQuickCheck,
-    NFKDQuickCheck,
-    NumericType,
-    SentenceBreak,
-    TrailCanonicalCombiningClass,
-    VerticalOrientation,
-    WordBreak,
+    BidiClass = 0x1000,
+    BidiPairedBracketType = 0x1015,
+    CanonicalCombiningClass = 0x1002,
+    DecompositionType = 0x1003,
+    EastAsianWidth = 0x1004,
+    GeneralCategory = 0x1005,
+    GraphemeClusterBreak = 0x1012,
+    HangulSyllableType = 0x100B,
+    IndicPositionalCategory = 0x1016,
+    IndicSyllabicCategory = 0x1017,
+    JoiningGroup = 0x1006,
+    JoiningType = 0x1007,
+    LineBreak = 0x1008,
+    LeadCanonicalCombiningClass = 0x1010,
+    NFCQuickCheck = 0x100E,
+    NFDQuickCheck = 0x100C,
+    NFKCQuickCheck = 0x100F,
+    NFKDQuickCheck = 0x100D,
+    NumericType = 0x1009,
+    SentenceBreak = 0x1013,
+    TrailCanonicalCombiningClass = 0x1011,
+    VerticalOrientation = 0x1018,
+    WordBreak = 0x1014,
 }
 
+
+// TODO: verify hunches that U_BLOCK_SEPARATOR is PPUCD Paragraph_Separator
+// TODO: ask why U_RIGHT_TO_LEFT_ARABIC is not in PPUCD
+// TODO: ask why PPUCD has ArabicLetter? Is that same as U_RIGHT_TO_LEFT_ARABIC?
```

Comment:
fwiw https://unicode.org/reports/tr9/#AL = Right To Left Arabic

Reply:
Done. That answered my question(s), thanks!  And when looking that up, I reminded myself that "AL" and "Arabic_Letter" are aliases when looking at PPUCD.

Reply:
FYI
> verify hunches that U_BLOCK_SEPARATOR is PPUCD Paragraph_Separator

Generally the answer to these kinds of questions is in https://github.com/unicode-org/icu/blob/main/icu4c/source/common/unicode/uchar.h

For example:
```
typedef enum UCharDirection {
    /*
     * Note: UCharDirection constants and their API comments are parsed by preparseucd.py.
     * It matches pairs of lines like
     *     / ** <Unicode 1..3-letter Bidi_Class value> comment... * /
     *     U_<[A-Z_]+> = <integer>,
     */
...
    /** B @stable ICU 2.0 */
    U_BLOCK_SEPARATOR             = 7,
```
In the comment, "B" is the short value alias. Now look in https://www.unicode.org/Public/UCD/latest/ucd/PropertyValueAliases.txt:
```
bc ; B                                ; Paragraph_Separator
```
My script that generates ppucd.txt parses not just UCD files but also uchar.h. Aside from ppucd.txt it also outputs a C++ header file that another tool includes for writing ICU's data for mapping between name strings and enum values.

> ask why U_RIGHT_TO_LEFT_ARABIC is not in PPUCD
> ask why PPUCD has ArabicLetter? Is that same as U_RIGHT_TO_LEFT_ARABIC?

See the "AL" in the comment:
```
    /** AL @stable ICU 2.0 */
    U_RIGHT_TO_LEFT_ARABIC        = 13,
```
vs. PropertyValueAliases.txt
```
bc ; AL                               ; Arabic_Letter
```

Reply:
The Python script generates this file:
https://github.com/unicode-org/icu/blob/main/tools/unicode/c/genprops/pnames_data.h

The genprops tool includes it, and the compiler resolves the enum constant names into their numeric values.

Would this, or something like it, be useful to you? We could beef up the Python script or the genprops code.